source/DSCResources/DSC_CMMaintenanceWindows/DSC_CMMaintenanceWindows.psm1

```diff
@@ -0,0 +1,520 @@
+$script:dscResourceCommonPath = Join-Path (Join-Path -Path (Split-Path -Parent -Path (Split-Path -Parent -Path $PsScriptRoot)) -ChildPath Modules) -ChildPath DscResource.Common
+$script:configMgrResourcehelper = Join-Path (Join-Path -Path (Split-Path -Parent -Path (Split-Path -Parent -Path $PsScriptRoot)) -ChildPath Modules) -ChildPath ConfigMgrCBDsc.ResourceHelper
+
+Import-Module -Name $script:dscResourceCommonPath
+Import-Module -Name $script:configMgrResourcehelper
+
+$script:localizedData = Get-LocalizedData -DefaultUICulture 'en-US'
+
+<#
+    .SYNOPSIS
+        This will return a hashtable of results.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER CollectionName
+        Specifies the collection name for the maintenance window.
+
+    .PARAMETER Name
+        Specifies the name for the maintenance window.
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $CollectionName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $Name
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    if (Get-CMCollection -Name $CollectionName)
+    {
+        $collect = 'Present'
+    }
+    else
+    {
+        $collect = 'Absent'
+    }
+
+    $windows = Get-CMMaintenanceWindow -CollectionName $CollectionName -MaintenanceWindowName $Name
```

Comment:
Get-CmMaintenanceWindow uses wildcards. Fix for better handling
$scubasteve = Get-CMMaintenanceWindow -CollectionName 'All Systems DP' -MaintenanceWindowName 'Monthly'
$freddy = $scubasteve | Where-Object -FilterScript {$_.name -eq 'Monthly'}